src/utils/optionsparser.cc

```diff
@@ -328,8 +325,14 @@ IntOption::ValueType IntOption::GetVal(const OptionsDict& dict) const {
   return dict.Get<ValueType>(GetName());
 }
 
-void IntOption::SetVal(OptionsDict* dict, const ValueType& val) const {
+bool IntOption::SetVal(OptionsDict* dict, const ValueType& val) const {
+  if (val < min_ || val > max_) {
+    std::cerr << "Flag '--" << GetLongFlag() << "' must be between "
```

Comment:
Instead of outputting it to stderr and reporting the error up (where it is used?), it's better to throw an Exception.
In this case it will also e.g. show error through uci protocol when bad parameters are attempted to be set through uci option command.

Reply:
I am a c++ exception noob.  I throw an exception now.  Should I be catching it somewhere or just let it crash and burn?  It crashes and burns right now!